### PR TITLE
[CI:DOCS] Man pages: refactor common options: --volume

### PR DESCRIPTION
**Pull request by edsantiago**
This one is a nightmare, because --volume has been edited
in four different files throughout the years (five if you
count podman-build, which I am not including in this PR).
Those edits have not always been done in sync.

The list of options was reordered 2022-06-28 by Giuseppe in #14734,
but only in podman-create and -run (not in podman-pod-*). No
explanation of why, but I'll assume he knew what he was doing,
and have accepted that for the reference copy.

There was also a big edit in #8519.

The "Propagation property...bind mounted" sentence first appeared
in pod-clone, in #14299 by cdoern, with no obvious source of where
it came from. I choose to include it in the reference copy.

The "**copy**" option seems to work in pod-create, so I'm including
it in the reference copy. Someone please yell loudly if this is
not the case.

The "disables SELinux separation for containers used in the build",
no idea, changed that to just "for the container/pod"

The "advanced users / overlay / upperdir / workdir" paragraph
makes zero sense to me, but hey, I assume it applies to all
the commands, so I put it in the reference copy.

Finally, there's still a mishmash of backticks, asterisks, underscores,
and even quotation marks. Someone is gonna have to perform major
cleanup on this one day, but at least it'll be in only one place.

Signed-off-by: Ed Santiago <santiago@redhat.com>
```release-note
None
```

### CODE REVIEW

File: docs/source/markdown/options/volume.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+Only the current <<container|pod>> can use a private volume.
+
+Note: Do not relabel system files and directories. Relabeling system content
+might cause other confined services on your machine to fail.  For these types
+of containers we recommend disabling SELinux separation.  The option
+**--security-opt label=disable** disables SELinux separation for the <<container|pod>>.
+For example if a user wanted to volume mount their entire home directory into a
+<<container|pod>>, they need to disable SELinux separation.
+
+	   $ podman <<subcommand>> --security-opt label=disable -v $HOME:/home/user fedora touch /home/user/file
```
Comment (edsantiago) [inline docs/source/markdown/options/volume.md:85]: And actually this entire paragraph (starting with "Note: Do not relabel") was present only in the podman-create and -run man pages, not in the pod-* ones. I choose to include it now in all. If this is wrong, please yell.

Oh, and, the `<<subcommand>>` on pod pages will show `podman create/run`, not `podman pod create/run`. I need to fix that, but it won't be today.

**Review by edsantiago — COMMENTED** [review]
I've made comments on what I think are the riskiest changes. There are tons more changes, but those are mostly backticks asterisks underscore type stuff. I think.

This is really impossible to review. My only suggestion is, ignore the man pages, read ONLY the new `volume.md` file, and think about whether every single word in it applies to each of the four man pages.

TIA.

File: docs/source/markdown/options/volume.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+in the host to the container. If no such named volume exists, Podman will
+create one. (Note when using the remote client, including Mac and Windows (excluding WSL2) machines, the volumes will be mounted from the remote server, not necessarily the client machine.)
+
+The _OPTIONS_ is a comma-separated list and can be: <sup>[[1]](#Footnote1)</sup>
+
+* **rw**|**ro**
+* **z**|**Z**
+* [**O**]
+* [**U**]
+* [**no**]**copy**
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply docs/source/markdown/options/volume.md:15]: Yes this should work.

File: docs/source/markdown/options/volume.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+objects on the shared volumes. The **z** option tells Podman that two <<containers|pods>>
+share the volume content. As a result, Podman labels the content with a shared
+content label. Shared volume labels allow all containers to read/write content.
+The **Z** option tells Podman to label the content with a private unshared label.
+Only the current <<container|pod>> can use a private volume.
+
+Note: Do not relabel system files and directories. Relabeling system content
+might cause other confined services on your machine to fail.  For these types
+of containers we recommend disabling SELinux separation.  The option
+**--security-opt label=disable** disables SELinux separation for the <<container|pod>>.
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply docs/source/markdown/options/volume.md:81]: Good, this is correct.

File: docs/source/markdown/options/volume.md (hunk 1 above)
Comment (rhatdan) [reply docs/source/markdown/options/volume.md:85]: It should be everywhere.

File: docs/source/markdown/options/volume.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+container storage in a separate directory. In overlay terms, the source
+directory will be the lower, and the container storage directory will be the
+upper. Modifications to the mount point are destroyed when the <<container|pod>>
+finishes executing, similar to a tmpfs mount point being unmounted.
+
+For advanced users, the **overlay** option also supports custom non-volatile
+**upperdir** and **workdir** for the overlay mount. Custom **upperdir** and
+**workdir** can be fully managed by the users themselves, and Podman will not
+remove it on lifecycle completion.
+Example **:O,upperdir=/some/upper,workdir=/some/work**
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply docs/source/markdown/options/volume.md:101]: Should be the same for all.

File: docs/source/markdown/options/volume.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+this behavior by specifying a volume mount propagation property. Making a
+volume shared mounts done under that volume inside the <<container|pod>> will be
+visible on host and vice versa. Making a volume **slave** enables only one
+way mount propagation and that is mounts done on host under that volume
+will be visible inside container but not the other way around. <sup>[[1]](#Footnote1)</sup>
+
+To control mount propagation property of a volume one can use the [**r**]**shared**,
+[**r**]**slave**, [**r**]**private** or the [**r**]**unbindable** propagation flag.
+Propagation property can be specified only for bind mounted volumes and not for
+internal volumes or named volumes. For mount propagation to work the source mount
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply docs/source/markdown/options/volume.md:135]: I believe this is correct for all.

**Comment by rhatdan** [comment]
LGTM
@giuseppe @mheon PTAL

File: docs/source/markdown/options/volume.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
+#### **--volume**, **-v**=*[[SOURCE-VOLUME|HOST-DIR:]CONTAINER-DIR[:OPTIONS]]*
+
+Create a bind mount. If `-v /HOST-DIR:/CONTAINER-DIR` is specified, Podman
+bind mounts `/HOST-DIR` in the host to `/CONTAINER-DIR` in the Podman
+container. Similarly, `-v SOURCE-VOLUME:/CONTAINER-DIR` will mount the volume
```
Comment (mheon) [inline docs/source/markdown/options/volume.md:5]: `mount the named volume from the host into the container`

Comment (mheon) [reply docs/source/markdown/options/volume.md:5]: Ah, no `podman build` manpage changes in here, OK. Ignore my earlier question.

File: docs/source/markdown/options/volume.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+* [**no**]**dev**
+* [**no**]**exec**
+* [**no**]**suid**
+* [**r**]**bind**
+* [**r**]**shared**|[**r**]**slave**|[**r**]**private**[**r**]**unbindable**
+
+The `CONTAINER-DIR` must be an absolute path such as `/src/docs`. The volume
+will be mounted into the container at this directory.
+
+Volumes may specify a source as well, as either a directory on the host
```
Comment (mheon) [inline docs/source/markdown/options/volume.md:25]: This feels like it should be combined with the first paragraph

File: docs/source/markdown/options/volume.md (hunk 6 above)
Comment (edsantiago) [reply docs/source/markdown/options/volume.md:5]: Thanks. Just to make ultra-sure, here is what the new diff looks like:
```diff
 Create a bind mount. If `-v /HOST-DIR:/CONTAINER-DIR` is specified, Podman
 bind mounts `/HOST-DIR` in the host to `/CONTAINER-DIR` in the Podman
-container. Similarly, `-v SOURCE-VOLUME:/CONTAINER-DIR` will mount the volume
-in the host to the container. If no such named volume exists, Podman will
+container. Similarly, `-v SOURCE-VOLUME:/CONTAINER-DIR` will mount the named
+volume from the host into the container. If no such named volume exists, Podman will
 create one. (Note ....
```

File: docs/source/markdown/options/volume.md (hunk 7 above)
Comment (edsantiago) [reply docs/source/markdown/options/volume.md:25]: Sorry to be really dense, but this is not my area of expertise. Could you give me a precise diff for me to apply, or defer this for a (much-needed) future man page cleanup? This option has taken a lot out of me, and I'm feeling nervous enough about it as it is. I don't trust my judgment enough to make this change on my own.

**Comment by edsantiago** [comment]
Re-pushed with @mheon's fixed wording and with a new `<<fullsubcommand>>` for the command example.

File: docs/source/markdown/options/volume.md (hunk 6 above)
Comment (mheon) [reply docs/source/markdown/options/volume.md]: LGTM

File: docs/source/markdown/options/volume.md (hunk 7 above)
Comment (mheon) [reply docs/source/markdown/options/volume.md:25]: Remove the first sentence, and graft the remainder of the paragraph onto the end of the first paragraph.

**Pull request merged**